DigitalLearningSolutions.Data/DataServices/CourseDataService.cs

```diff
@@ -181,5 +183,57 @@ public IEnumerable<CourseStatistics> GetCourseStatisticsAtCentreForCategoryId(in
                 new { centreId, categoryId }
             );
         }
+
+        public IEnumerable<DelegateCourseInfo> GetDelegateCoursesInfo(int delegateId)
+        {
+            return connection.Query<DelegateCourseInfo>(
+                @"SELECT
+                        cu.CustomisationID AS CustomisationId,
+                        ap.ApplicationName,
+                        cu.CustomisationName,
+                        (SELECT CONCAT(Forename, ' ', Surname)
+                         FROM AdminUsers
+                         WHERE AdminID = pr.SupervisorAdminId
+                        ) AS Supervisor,
+                        pr.FirstSubmittedTime AS Enrolled,
+                        pr.SubmittedTime AS LastUpdated,
+                        pr.CompleteByDate AS CompleteBy,
+                        pr.Completed AS Completed,
+                        pr.Evaluated AS Evaluated,
+                        pr.EnrollmentMethodID AS EnrolmentMethodId,
+                        pr.LoginCount,
+                        pr.Duration AS LearningTime,
+                        pr.DiagnosticScore,
+                        cu.IsAssessed,
+                        pr.Answer1,
+                        pr.Answer2,
+                        pr.Answer3
+                    FROM Customisations cu
+                    INNER JOIN Applications ap ON ap.ApplicationID = cu.ApplicationID
+                    INNER JOIN Progress pr ON pr.CustomisationID = cu.CustomisationID
+                    WHERE pr.CandidateID = @delegateId
+                        AND pr.RemovedDate IS NULL
+                        AND pr.SystemRefreshed = 0",
+                new { delegateId }
+            );
+        }
+
+        public (int totalAttempts, int attemptsPassed) GetDelegateCourseAttemptStats(
+            int delegateId,
+            int customisationId
+        )
+        {
+            return connection.QueryFirstOrDefault<(int, int)>(
+                @"SELECT COUNT(aa.Status) AS TotalAttempts,
+                        COUNT(CASE WHEN aa.Status=1 THEN 1 END) AS AttemptsPassed

```

Comment:
I wonder if it makes sense pulling these two numbers together with other course data?

Reply:
I thought it would make the query too complicated. Also these are not needed in many cases any where IsAssessed=0.

Reply:
I think there is a bit more refactoring needed in that case (also see my comment on the controller). If these aren't needed because IsAssessed=0, why are we still always calling it?